### PR TITLE
Animate NSLayoutConstraint.constant using signals

### PR DESCRIPTION
**Pull request by indragiek**
First attempt at implementing #4.

### CODE REVIEW

File: ReactiveAnimation/Constraint.swift [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+//
+//  Created by Indragie on 3/25/15.
+//  Copyright (c) 2015 ReactiveCocoa. All rights reserved.
+//
+
+import Foundation
+import ReactiveCocoa
+
+// Wraps an NSLayoutConstraint for animating its constant using signals.
+public struct RANConstraint {
```
Comment (indragiek) [inline ReactiveAnimation/Constraint.swift:13]: Should this be named `Constraint` instead? I named it this way to be consistent with `RANView`.

File: ReactiveAnimation/Constraint.swift [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+	private weak var layoutConstraint: NSLayoutConstraint?
+	private let willDealloc: SignalProducer<(), NoError>
+	
+	// Creates a wrapper for the given layout constraint.
+	public init(_ layoutConstraint: NSLayoutConstraint) {
+		self.layoutConstraint = layoutConstraint
+		self.willDealloc = layoutConstraint.rac_willDeallocSignal().toSignalProducer()
```
Comment (indragiek) [inline ReactiveAnimation/Constraint.swift:20]: Ideally there would be a better way to determine whether a constraint becomes inactive than deallocation. `NSLayoutConstraint` has an `active` property, but the docs don't explicitly say its KVO-able.

**Comment by jspahrsummers** [comment]
Looks good upon a quick skim. :sparkles:

@indragiek Realistically, I probably won't have a whole lot of time to review every PR on here, but I would love to see the project continue—so please feel free to self-merge, if that's helpful. :pray:

Of course, if you ever specifically want me to chime in, mention me and I'll prioritize it.

**Comment by indragiek** [comment]
@jspahrsummers Thanks, I'll try my best to keep this going when I have time. I'll see what I can do about putting together some examples as well.

Just a question with regards to what's in this PR: do you think it would be an expected behaviour to have signals bound to constraints even after they have been deactivated (i.e. removed from their view)?

**Comment by jspahrsummers** [comment]
> do you think it would be an expected behaviour to have signals bound to constraints even after they have been deactivated (i.e. removed from their view)?

Yeah, I think so. The binding should remain in _effect_, even though it won't _affect_ what displays on screen.

Only when the constraint is deallocated should the binding cancel, IMO.

**Comment by indragiek** [comment]
@jspahrsummers :+1: I'll come up with an addition to the example app for this just to make sure everything works as intended and self merge.

**Pull request closed without merging**